src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -635,6 +652,24 @@ private boolean activatePresetResult() {
 	 * @throws Exception Throw exception when fail to get JsonNode
 	 */
 	private boolean activatePreset(String presetName) throws Exception {
+		// Check if preset is exist
+		Map<Object, Object> presetParam = new HashMap<>();
+		presetParam.put(BarcoE2Constant.ID, BarcoE2Constant.LIST_ALL_DESTINATION_FOR_PRESET);
+		JsonNode allPreset = requestByMethod(BarcoE2Constant.METHOD_LIST_DESTINATIONS_FOR_PRESET, presetParam);
+		if (allPreset == null) {
+			throw new ResourceNotReachableException("Fail to get list of preset");
+		}
+		boolean presetStatus = false;
+		for (int i = 0; i < allPreset.size(); i++) {
+			if (allPreset.get(i).get(BarcoE2Constant.NAME) != null && allPreset.get(i).get(BarcoE2Constant.NAME).asText().equals(presetName)) {
```

Comment:
is there a chance for an NPE with allPreset.get(i).get.... ?
Also .get(BarcoE2Constant.NAME) can be extracted into variable